json/classic/src/test/java/ch/qos/logback/contrib/json/classic/JsonLayoutTest.java

```diff
@@ -0,0 +1,149 @@
+
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.addMap("key1", true, emptyMap, map);
+        jsonLayout.addMap("key2", true, mapWithData, map);
+        jsonLayout.addMap("key3", true, mapWithArrayValue, map);
+        jsonLayout.addMap("key4", false, mapWithArrayValue, map);
+
+        assertTrue(map.size() == 2);
+        assertFalse(map.containsKey("key1"));
+        assertEquals(mapWithData, map.get("key2"));
+        assertEquals(mapWithArrayValue, map.get("key3"));
+        assertFalse(map.containsKey("key4"));
+    }
+
+    @Test
+    public void jsonLayout() throws Exception {
+        configure("src/test/input/json/jsonLayout.xml");
+        String loggerName = "STR_LIST";
+        String message = "Info message";
+        String debugMessage = "Debug message";
+        Logger logger = context.getLogger(loggerName);
+        logger.info("Test");
+
+        ILoggingEvent event = new LoggingEvent("my.class.name", logger, Level.INFO, message, null, null);
+
+        JsonLayout jsonLayout = new JsonLayout();
+        jsonLayout.setContext(context);
+        String log = jsonLayout.doLayout(event);
+
+        assertTimestamp(log);
+        assertTrue(log.contains(String.format("%s=%s", JsonLayout.LEVEL_ATTR_NAME, Level.INFO)));
+        assertTrue(log.contains(String.format("%s=%s", JsonLayout.THREAD_ATTR_NAME, "main")));
+        assertTrue(log.contains(String.format("%s=%s", JsonLayout.LOGGER_ATTR_NAME, loggerName)));
+        assertTrue(log.contains(String.format("%s=%s", JsonLayout.FORMATTED_MESSAGE_ATTR_NAME, message)));
+
+        jsonLayout.setIncludeContextName(true);
+        jsonLayout.setIncludeMDC(true);
+        jsonLayout.setIncludeLoggerName(true);
+        jsonLayout.setIncludeException(true);
+        jsonLayout.setIncludeMessage(true);
+
+        RuntimeException exception = new RuntimeException("Exception");
+        ILoggingEvent eventWithException = new LoggingEvent("my.class.name", logger, Level.DEBUG, debugMessage, exception, null);
+        String logWithException = jsonLayout.doLayout(eventWithException);
+
+        assertTimestamp(logWithException);
+        assertTrue(logWithException.contains(String.format("%s=%s", JsonLayout.LEVEL_ATTR_NAME, Level.DEBUG)));
+        assertTrue(logWithException.contains(String.format("%s=%s", JsonLayout.LOGGER_ATTR_NAME, loggerName)));
+        assertTrue(logWithException.contains(String.format("%s=%s", JsonLayout.FORMATTED_MESSAGE_ATTR_NAME, debugMessage)));
+        assertTrue(logWithException.contains(String.format("%s=%s", JsonLayout.MESSAGE_ATTR_NAME, debugMessage)));
+        assertTrue(logWithException.contains(String.format("%s=%s", JsonLayout.EXCEPTION_ATTR_NAME, exception.toString())));
+    }
+
+    private void assertTimestamp(String log) {
+        int timestampStart = log.indexOf(JsonLayout.TIMESTAMP_ATTR_NAME) + JsonLayout.TIMESTAMP_ATTR_NAME.length() + 1;
```

Comment:
`log.indexOf(JsonLayout.TIMESTAMP_ATTR_NAME)` could return -1, yet this line could still yield a valid start index, which could potentially lead to a valid timestamp and thus false positive.